managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1545,21 +1546,21 @@ public void operationFailed(MetaStoreException e) {
                 }
             };
 
-            updateLedgersListAfterRollover(cb);
+            updateLedgersListAfterRollover(cb, getManagedLedgerInfo(ledgersTmp));
         }
     }
-
-    private void updateLedgersListAfterRollover(MetaStoreCallback<Void> callback) {
+    private void updateLedgersListAfterRollover(MetaStoreCallback<Void> callback, ManagedLedgerInfo mlInfo) {
         if (!metadataMutex.tryLock()) {
             // Defer update for later
-            scheduledExecutor.schedule(() -> updateLedgersListAfterRollover(callback), 100, TimeUnit.MILLISECONDS);
+            scheduledExecutor.schedule(() -> updateLedgersListAfterRollover(callback, mlInfo),
```

Comment:
I think we are not handling this condition well:

If the process runs at this line, we continue to retry later. But at the same time, the `ledgers` is changed. However, we still use old data to update. I'm not sure if it is a big problem. But I think we have to clarify it.
I think I have to give a change request.  @lordcheng10  Could you help to confirm it?

Reply:
> Why do we need to create a new method?
In order to avoid affecting other logic that uses getManagedLedgerInfo, a getManagedLedgerInfo method is added here.
The new method has made some changes again, please review again, thanks!

Reply:
> I think we are not handling this condition well:
If the process runs at this line, we continue to retry later. But at the same time, the ledgers is changed. However, we still use old data to update. I'm not sure if it is a big problem. But I think we have to clarify it.
I think I have to give a change request. @lordcheng10 Could you help to confirm it?



This is eventual consistency, I don't think it's a problem.

Reply:
@lordcheng10  
Yes, I also don't think it's a big problem, but maybe we can leave this comment here. Perhaps someone can find some things that will be affected by this.

Reply:
@lordcheng10 
I'm not very sure about that, maybe we need more eyes on it.